metarecord/views/function.py

```diff
@@ -51,24 +56,73 @@ def validate_function_id(self, value):
             raise exceptions.ValidationError(_('Function ID %s already exists.') % value)
         return value
 
+    def validate(self, data):
+        if self.partial:
+            if 'state' not in data:
+                raise exceptions.ValidationError({'state': self.error_messages['required']})
+            self.check_state_change(self.instance.state, data['state'])
```

Comment:
I find this idiom curious, but it might be pythonic. That is, having a function which either does nothing (returns) or raises one of several possible exceptions. Will merge anyway, but just wondering if this could be done something eg. thusly:
```
if not self.is_valid_state(...):
    raise ValidationError
if not self.user_has_permission(...):
    raise PermissionError
```
Just thinking out loud.